test/testcookies.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+   This software is in the public domain, furnished "as is", without technical
+   support, and with no warranty, express or implied, as to its usefulness for
+   any purpose.
+*/
+
+#include <QtTest>
+
+#include "libsync/cookiejar.h"
+
+using namespace OCC;
+
+class TestCookies : public QObject
+{
+    Q_OBJECT
+
+private slots:
+    void testCookies()
+    {
+        QTemporaryDir tmp;
+        const QString nonexistingPath = tmp.filePath("someNonexistingDir/test.db");
+        QNetworkCookie cookieA = QNetworkCookie("foo", "bar");
+        // tomorrow rounded
+        cookieA.setExpirationDate(QDateTime(QDateTime::currentDateTimeUtc().addDays(1).date()));
+        const QList<QNetworkCookie> cookies = {cookieA, QNetworkCookie("foo2", "bar")};
+        CookieJar jar;
+        jar.setAllCookies(cookies);
+        QCOMPARE(cookies, jar.allCookies());
+        QVERIFY(jar.save(tmp.filePath("test.db")));
+        QVERIFY(jar.save(nonexistingPath));
+
```

Comment:
At first it was a bit confusing to me why you saved to two different paths and only verified loading only in one of them afterwards... maybe leave a comment what you are testing exactly.